stackstate_checks_base/stackstate_checks/base/checks/base_v2.py

```diff
@@ -0,0 +1,330 @@
+import copy
+import json
+import traceback
+
+from schematics import Model
+from typing import Any, Dict, Optional, Union, TypeVar
+
+from .base import AgentCheck
+from ..utils.common import to_string
+from ..utils.health_api import HealthApi, HealthStream
+from ..utils.state_api import StateApi
+from ..utils.transactional_api import TransactionApi
+
+try:
+    import topology
+
+    using_stub_topology = False
+except ImportError:
+    from ..stubs import topology
+
+    using_stub_topology = True
+
+_InstanceType = TypeVar('_InstanceType', Model, Dict[str, Any])
+
+DEFAULT_RESULT = to_string(b'')
+
+
+class CheckError(Exception):
+    """
+    Agent error that needs to be converted to Critical Service Check
+    """
+
+    def to_string(self):
+        # type: () -> str
+        return "check error: {}".format(self.message)
+
+
+class CheckMixin(object):
```

Comment:
Maybe this is CheckSetupMixin? If I understand this mixin is used only to declare the `setup` method, right?